airflow/api/common/trigger_dag.py

```diff
@@ -51,7 +51,11 @@ def _trigger_dag(
     if dag is None or dag_id not in dag_bag.dags:
         raise DagNotFound(f"Dag id {dag_id} not found")
 
-    execution_date = execution_date if execution_date else timezone.utcnow()
+    if execution_date is None:
+        if conf and "logical_date" in conf:
+            execution_date = conf["logical_date"]
```

Comment:
I don’t recall specifying `logical_date` in trigger config being a valid usage? Why is this added?

Reply:
In this [tutorial](https://airflow.apache.org/docs/apache-airflow/1.10.1/tutorial.html), default_args doesn't provide logical_date, but the issuer was assigning execution_date to default_args. 

I think we should check validation of using default_args.

Reply:
Maybe we can set whitelist.

Reply:
https://github.com/apache/airflow/blob/401fc57e8ba1dddb041e0d777bb0277a09f227db/airflow/models/dag.py#L295 
The comment describe, default_arg values can be conflict with operator's arguments, it will follow to operator's argument. 
Should `execution_date` argument follow this rule?

Reply:
`default_args` != `dag.conf`

1. https://airflow.apache.org/docs/apache-airflow/stable/dag-run.html#passing-parameters-when-triggering-dags
2. https://airflow.apache.org/docs/apache-airflow/stable/tutorial/fundamentals.html#tasks

Reply:
`dag_run.conf` use in Templates, and `default_args` parameters propagate to tasks arguments if it not set explicitly.

Reply:
Thank you for the explanation. 
Than, the issuer might have used `default_args` to parse `logical_date`

Reply:
I've change conf to default_args.

Reply:
And this also incorrect assumption.

`default_args` designed for reuse same parameters across Tasks within the single DAG, and it user-defined on code level. This arguments do not changed by their nature in runtime, except selected arguments:

1. `params` - [Params](https://airflow.apache.org/docs/apache-airflow/stable/concepts/params.html#params) could defined on DAG and Task level, so it resolved during DAG initialisation
2. `start_date` and `end_date` - https://airflow.apache.org/docs/apache-airflow/stable/faq.html#what-s-the-deal-with-start-date

That is all exceptions. 


`default_args` doesn't not contain `logical_date` / `execution_date` unless user define it, and this might happen if required propagate this arguments to [TriggerDagRunOperator](https://airflow.apache.org/docs/apache-airflow/stable/_api/airflow/operators/trigger_dagrun/index.html#airflow.operators.trigger_dagrun.TriggerDagRunOperator) or [ExternalTaskSensor](https://airflow.apache.org/docs/apache-airflow/stable/_api/airflow/sensors/external_task/index.html#airflow.sensors.external_task.ExternalTaskSensor) without explicit define it during initialisation. But even in this case most probably it would be something like:

```python

with DAG(
    ..., 
    default_args={
        "execution_date": "{{ logical_date }}"
    },
) as dag:
    ...
```

Reply:
Okay. 
According to the issue, the issuer triggered the dag with config. so I assumed it was triggered with execution_date which is set by default_args.
but not I think there is another reason of this issue. I'll look it up more.